File: frontend/src/services/onboardingService.js
import apiClient from './api';

export const getTasks = () => {
  return apiClient.get('/onboarding/tasks');
};

export const completeTask = (id) => {
  return apiClient.post(`/onboarding/tasks/${id}/complete`);
};

export const getAssets = () => {
  return apiClient.get('/onboarding/assets');
};

export const logSignature = (dto) => {
  return apiClient.post('/onboarding/signature', dto);
};

File: backend/src/main/java/com/awais/hr/module/onboarding/service/OnboardingService.java
package com.awais.hr.module.onboarding.service;

import com.awais.hr.module.onboarding.dto.PolicySignatureRequestDTO;
import java.util.List;
import java.util.Map;

public interface OnboardingService {
    List<Map<String, Object>> getTasks(String email);
    void completeTask(String id);
    List<Map<String, Object>> getAssets(String email);
    void logSignature(PolicySignatureRequestDTO dto);
}
